src/clusterer/clusterer.py

```
@@ -1,8 +1,71 @@
-def main() -> int:
-    """
-    Dummy function for the setup
+import json
+import pathlib
+from typing import Optional, Any
+from typing_extensions import Self  # <3.11
+
+import numpy as np
+import numpy.typing as npt
 
-    :return: A dummy value.
-    :rtype: int
+
+class DataLoader:
+    """
+    Returns the content of the data source.
     """
-    return 1
+
+    def __init__(self, data: npt.NDArray[Any], filename: str) -> None:  # PEP-0484
+        self.data: npt.NDArray[Any] = data
+        self.filename = filename
+
+    @classmethod
+    def from_numpy(cls, source: str, suffix: str = ".npy") -> Optional[Self]:
+        """
+        Create a DataLoader instance from a file with .NPY suffix.
+
+        :param source: Data in numpy array format.
+        :type source: str
+        :param suffix: extension of the supported file.
+        :type suffix: str
+        :return: the loaded numpy content,
+        :rtype: DataLoader
+        """
+        if not pathlib.Path(source).suffix == suffix:
+            raise Exception("Input file does not have correct suffix.")
+
+        try:
+            file_content = np.load(source)
+
+        except FileNotFoundError as err:
+            print(repr(err))
+            return None
```

it is probably better to reraise the exception or at least use logger to log the exception. The exit code of the main function should reflect the type of the error.

---

Agree.

---

ping

---

Normally I wouldn't handle it this way, but for this project I raised an exception again.